lib/filewatch/watched_file.rb

```diff
@@ -89,8 +89,27 @@ def file_seek(amount, whence = IO::SEEK_SET)
     end
 
     def file_read(amount)
-      set_accessed_at
-      @file.sysread(amount)
+      #debug "*** file_read #{path}"
+      cc = 3        # max attempts
+      dt = 1.0/128  # delay beetwen attempts
+      loop do
+        set_accessed_at
+        buf = @file.sysread(amount)
+        # return if no zero byte inside
+        return buf unless zc = buf.index("\0")
+        # update amount to read
+        amount = buf.bytesize
```

Comment:
`amount` here has multiple meanings (amount requested vs amount retrieved), and the mutation of the input parameter combined with the retry pattern means that on each subsequent attempt we will read potentially _fewer_ bytes than initially requested. This is rather opaque and can present surprises later on that would be hard to debug, so we try to avoid overloading variables like this.